packages/components/src/tabs/index.tsx

```diff
@@ -154,26 +159,50 @@ function Tabs( {
 		setSelectedId,
 	] );
 
-	// In controlled mode, make sure browser focus follows the selected tab if
-	// the selection is changed while a tab is already being focused.
-	useLayoutEffect( () => {
-		if ( ! isControlled || ! selectOnMove ) {
+	const previousSelectedId = usePrevious( selectedId );
+
+	useEffect( () => {
+		if ( ! isControlled ) {
 			return;
 		}
 		const currentItem = items.find( ( item ) => item.id === selectedId );
 		const activeElement = currentItem?.element?.ownerDocument.activeElement;
-		const tabsHasFocus = items.some( ( item ) => {
-			return activeElement && activeElement === item.element;
-		} );
 
-		if (
+		const tabsHasFocus =
```

Comment:
I'd go for another early return here, which tends to be easier on cognitive load, but YMMV. (No strong opinion)

```diff
diff --git a/packages/components/src/tabs/index.tsx b/packages/components/src/tabs/index.tsx
index cf6443c8bd..bf857526dc 100644
--- a/packages/components/src/tabs/index.tsx
+++ b/packages/components/src/tabs/index.tsx
@@ -168,22 +168,20 @@ function Tabs( {
 		const currentItem = items.find( ( item ) => item.id === selectedId );
 		const activeElement = currentItem?.element?.ownerDocument.activeElement;
 
-		const tabsHasFocus =
-			activeElement &&
-			items.some( ( item ) => {
-				return activeElement === item.element;
-			} );
+		if (
+			! activeElement ||
+			! items.some( ( item ) => activeElement === item.element )
+		) {
+			return; // Return early if no tabs are focused.
+		}
+
 		const previousSelectedTabHadFocus =
 			typeof previousSelectedId === 'string' &&
 			previousSelectedId === activeElement?.id;
 
 		// If the previously selected tab had focus when the selection changed,
 		// move focus to the newly selected tab.
-		if (
-			tabsHasFocus &&
-			previousSelectedTabHadFocus &&
-			selectedId !== activeElement.id
-		) {
+		if ( previousSelectedTabHadFocus && selectedId !== activeElement.id ) {
 			move( selectedId );
 			return;
 		}
@@ -192,7 +190,7 @@ function Tabs( {
 		// selection changed, update the activeId to the currently focused tab.
 		// The activeId controls how arrow key navigation behaves. Keeping them
 		// in sync avoids confusion when navigating tabs with the keyboard.
-		if ( tabsHasFocus && ! previousSelectedTabHadFocus ) {
+		if ( ! previousSelectedTabHadFocus ) {
 			setActiveId( activeElement.id );
 		}
 	}, [
```

Reply:
oh, good call I like that a lot, thanks!